source/extensions/filters/http/cache/cache_filter.cc

```diff
@@ -30,11 +31,12 @@ struct CacheResponseCodeDetailValues {
 
 using CacheResponseCodeDetails = ConstSingleton<CacheResponseCodeDetailValues>;
 
-CacheFilter::CacheFilter(const envoy::extensions::filters::http::cache::v3::CacheConfig& config,
-                         const std::string&, Stats::Scope&, TimeSource& time_source,
-                         HttpCache& http_cache)
-    : time_source_(time_source), cache_(http_cache),
-      vary_allow_list_(config.allowed_vary_headers()) {}
+bool is_route_config_init_ = false;
```

Comment:
This does not look thread-safe at first glance, and that might be the tip of the iceberg of issues with this. Can you add code comments detailing the lifetime and scope of the route config init and how it relates to (say) having multiple tenants using the cache in the same process?


Update: I see this is also the name of a non-static member variable. Maybe you meant to just use the member variable and not a global? Can you just delete the above line?